examples/pixelcnn/train_test.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for flax.examples.pixelcnn.train."""
+
+import pathlib
+import tempfile
+
+from absl.testing import absltest
+
+import tensorflow_datasets as tfds
+
+from configs import testing as testing_lib
+import train as pixelcnn_train
+
+
+class PixelCnnTest(absltest.TestCase):
+  """Test cases for train."""
+
+  def test_train_and_evaluate(self):
+    """Tests pixelcnn training and evaluation loop."""
+    # Create a temporary directory where tensorboard metrics are written.
+    model_dir = tempfile.mkdtemp()
+
+    # Go two directories up to the root of the flax directory.
+    flax_root_dir = pathlib.Path(__file__).parents[2]
+    data_dir = str(flax_root_dir) + '/.tfds/metadata'
```

Comment:
Wouldn't it be better to read this from local path `'./.tfds/metadata'` ?

In case tests are run in Github action etc : it's the same, since they're run from the Flax root directory.

In case examples are downloaded independently / forked : There wouldn't be a `../../.tfds/metadata` directory; the test should rather complain with an error message how to get these files - not sure if this is an important usecase, though.

Reply:
So the idea behind this having it this was to have:
- TFDS metadata is downloaded exclusively for these unit tests (we download it automatically in `run_all_tests.sh`) and we place it under flax/.tfds/.. If somebody runs a test individually, they would need to run `download_tfds_metadata` script. Since the downloaded content is specific to Flax, I thought may be it would be best to have in the same directory locality i.e somewhere under flax.

I think we can move this discussion to another forum may be as this will need to be changed (if we agree on something) for other unit tests as well.

Reply:
SGTM, let's move this to a separate discussion.